[2024-12-08T14:39:05.837+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2024-12-08T14:39:05.852+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: sparking_flow.python_job manual__2024-12-08T14:39:04.163267+00:00 [queued]>
[2024-12-08T14:39:05.862+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: sparking_flow.python_job manual__2024-12-08T14:39:04.163267+00:00 [queued]>
[2024-12-08T14:39:05.864+0000] {taskinstance.py:2866} INFO - Starting attempt 1 of 1
[2024-12-08T14:39:05.874+0000] {taskinstance.py:2889} INFO - Executing <Task(SparkSubmitOperator): python_job> on 2024-12-08 14:39:04.163267+00:00
[2024-12-08T14:39:05.880+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'sparking_flow', 'python_job', 'manual__2024-12-08T14:39:04.163267+00:00', '--job-id', '166', '--raw', '--subdir', 'DAGS_FOLDER/spark_***.py', '--cfg-path', '/tmp/tmpsvml_q5_']
[2024-12-08T14:39:05.882+0000] {logging_mixin.py:190} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70 DeprecationWarning: This process (pid=185) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2024-12-08T14:39:05.883+0000] {standard_task_runner.py:105} INFO - Job 166: Subtask python_job
[2024-12-08T14:39:05.883+0000] {standard_task_runner.py:72} INFO - Started process 192 to run task
[2024-12-08T14:39:05.929+0000] {task_command.py:467} INFO - Running <TaskInstance: sparking_flow.python_job manual__2024-12-08T14:39:04.163267+00:00 [running]> on host 2a892a3469ec
[2024-12-08T14:39:06.007+0000] {taskinstance.py:3132} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='Yusuf Ganiyu' AIRFLOW_CTX_DAG_ID='sparking_flow' AIRFLOW_CTX_TASK_ID='python_job' AIRFLOW_CTX_EXECUTION_DATE='2024-12-08T14:39:04.163267+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-12-08T14:39:04.163267+00:00'
[2024-12-08T14:39:06.008+0000] {taskinstance.py:731} INFO - ::endgroup::
[2024-12-08T14:39:06.031+0000] {base.py:84} INFO - Retrieving connection 'spark-conn'
[2024-12-08T14:39:06.032+0000] {spark_submit.py:473} INFO - Spark-Submit cmd: spark-submit --master spark://sparkingflow-spark-master-1:7077 --name arrow-spark --deploy-mode client jobs/python/wordcountjob.py
[2024-12-08T14:39:11.762+0000] {spark_submit.py:634} INFO - 24/12/08 14:39:11 INFO SparkContext: Running Spark version 3.5.3
[2024-12-08T14:39:11.765+0000] {spark_submit.py:634} INFO - 24/12/08 14:39:11 INFO SparkContext: OS info Linux, 6.10.14-linuxkit, aarch64
[2024-12-08T14:39:11.766+0000] {spark_submit.py:634} INFO - 24/12/08 14:39:11 INFO SparkContext: Java version 17.0.13
[2024-12-08T14:39:11.851+0000] {spark_submit.py:634} INFO - 24/12/08 14:39:11 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2024-12-08T14:39:12.035+0000] {spark_submit.py:634} INFO - 24/12/08 14:39:12 INFO ResourceUtils: ==============================================================
[2024-12-08T14:39:12.043+0000] {spark_submit.py:634} INFO - 24/12/08 14:39:12 INFO ResourceUtils: No custom resources configured for spark.driver.
[2024-12-08T14:39:12.044+0000] {spark_submit.py:634} INFO - 24/12/08 14:39:12 INFO ResourceUtils: ==============================================================
[2024-12-08T14:39:12.045+0000] {spark_submit.py:634} INFO - 24/12/08 14:39:12 INFO SparkContext: Submitted application: arrow-spark
[2024-12-08T14:39:12.091+0000] {spark_submit.py:634} INFO - 24/12/08 14:39:12 INFO ResourceProfile: Default ResourceProfile created, executor resources: Map(memory -> name: memory, amount: 1024, script: , vendor: , offHeap -> name: offHeap, amount: 0, script: , vendor: ), task resources: Map(cpus -> name: cpus, amount: 1.0)
[2024-12-08T14:39:12.105+0000] {spark_submit.py:634} INFO - 24/12/08 14:39:12 INFO ResourceProfile: Limiting resource is cpu
[2024-12-08T14:39:12.111+0000] {spark_submit.py:634} INFO - 24/12/08 14:39:12 INFO ResourceProfileManager: Added ResourceProfile id: 0
[2024-12-08T14:39:12.235+0000] {spark_submit.py:634} INFO - 24/12/08 14:39:12 INFO SecurityManager: Changing view acls to: ***
[2024-12-08T14:39:12.237+0000] {spark_submit.py:634} INFO - 24/12/08 14:39:12 INFO SecurityManager: Changing modify acls to: ***
[2024-12-08T14:39:12.239+0000] {spark_submit.py:634} INFO - 24/12/08 14:39:12 INFO SecurityManager: Changing view acls groups to:
[2024-12-08T14:39:12.242+0000] {spark_submit.py:634} INFO - 24/12/08 14:39:12 INFO SecurityManager: Changing modify acls groups to:
[2024-12-08T14:39:12.254+0000] {spark_submit.py:634} INFO - 24/12/08 14:39:12 INFO SecurityManager: SecurityManager: authentication disabled; ui acls disabled; users with view permissions: ***; groups with view permissions: EMPTY; users with modify permissions: ***; groups with modify permissions: EMPTY
[2024-12-08T14:39:12.701+0000] {spark_submit.py:634} INFO - 24/12/08 14:39:12 INFO Utils: Successfully started service 'sparkDriver' on port 41307.
[2024-12-08T14:39:12.770+0000] {spark_submit.py:634} INFO - 24/12/08 14:39:12 INFO SparkEnv: Registering MapOutputTracker
[2024-12-08T14:39:12.851+0000] {spark_submit.py:634} INFO - 24/12/08 14:39:12 INFO SparkEnv: Registering BlockManagerMaster
[2024-12-08T14:39:12.911+0000] {spark_submit.py:634} INFO - 24/12/08 14:39:12 INFO BlockManagerMasterEndpoint: Using org.apache.spark.storage.DefaultTopologyMapper for getting topology information
[2024-12-08T14:39:12.912+0000] {spark_submit.py:634} INFO - 24/12/08 14:39:12 INFO BlockManagerMasterEndpoint: BlockManagerMasterEndpoint up
[2024-12-08T14:39:12.919+0000] {spark_submit.py:634} INFO - 24/12/08 14:39:12 INFO SparkEnv: Registering BlockManagerMasterHeartbeat
[2024-12-08T14:39:12.964+0000] {spark_submit.py:634} INFO - 24/12/08 14:39:12 INFO DiskBlockManager: Created local directory at /tmp/blockmgr-90cb0d0d-befb-42fe-8dc1-57c5009b1578
[2024-12-08T14:39:12.992+0000] {spark_submit.py:634} INFO - 24/12/08 14:39:12 INFO MemoryStore: MemoryStore started with capacity 434.4 MiB
[2024-12-08T14:39:13.024+0000] {spark_submit.py:634} INFO - 24/12/08 14:39:13 INFO SparkEnv: Registering OutputCommitCoordinator
[2024-12-08T14:39:13.251+0000] {spark_submit.py:634} INFO - 24/12/08 14:39:13 INFO JettyUtils: Start Jetty 0.0.0.0:4040 for SparkUI
[2024-12-08T14:39:13.338+0000] {spark_submit.py:634} INFO - 24/12/08 14:39:13 WARN Utils: Service 'SparkUI' could not bind on port 4040. Attempting port 4041.
[2024-12-08T14:39:13.339+0000] {spark_submit.py:634} INFO - 24/12/08 14:39:13 WARN Utils: Service 'SparkUI' could not bind on port 4041. Attempting port 4042.
[2024-12-08T14:39:13.348+0000] {spark_submit.py:634} INFO - 24/12/08 14:39:13 INFO Utils: Successfully started service 'SparkUI' on port 4042.
[2024-12-08T14:39:13.629+0000] {spark_submit.py:634} INFO - 24/12/08 14:39:13 INFO StandaloneAppClient$ClientEndpoint: Connecting to master spark://sparkingflow-spark-master-1:7077...
[2024-12-08T14:39:13.722+0000] {spark_submit.py:634} INFO - 24/12/08 14:39:13 INFO TransportClientFactory: Successfully created connection to sparkingflow-spark-master-1/172.18.0.5:7077 after 50 ms (0 ms spent in bootstraps)
[2024-12-08T14:39:14.070+0000] {spark_submit.py:634} INFO - 24/12/08 14:39:14 INFO StandaloneSchedulerBackend: Connected to Spark cluster with app ID app-20241208143914-0001
[2024-12-08T14:39:14.137+0000] {spark_submit.py:634} INFO - 24/12/08 14:39:14 INFO Utils: Successfully started service 'org.apache.spark.network.netty.NettyBlockTransferService' on port 38023.
[2024-12-08T14:39:14.138+0000] {spark_submit.py:634} INFO - 24/12/08 14:39:14 INFO NettyBlockTransferService: Server created on 2a892a3469ec:38023
[2024-12-08T14:39:14.162+0000] {spark_submit.py:634} INFO - 24/12/08 14:39:14 INFO BlockManager: Using org.apache.spark.storage.RandomBlockReplicationPolicy for block replication policy
[2024-12-08T14:39:14.199+0000] {spark_submit.py:634} INFO - 24/12/08 14:39:14 INFO BlockManagerMaster: Registering BlockManager BlockManagerId(driver, 2a892a3469ec, 38023, None)
[2024-12-08T14:39:14.276+0000] {spark_submit.py:634} INFO - 24/12/08 14:39:14 INFO BlockManagerMasterEndpoint: Registering block manager 2a892a3469ec:38023 with 434.4 MiB RAM, BlockManagerId(driver, 2a892a3469ec, 38023, None)
[2024-12-08T14:39:14.302+0000] {spark_submit.py:634} INFO - 24/12/08 14:39:14 INFO BlockManagerMaster: Registered BlockManager BlockManagerId(driver, 2a892a3469ec, 38023, None)
[2024-12-08T14:39:14.318+0000] {spark_submit.py:634} INFO - 24/12/08 14:39:14 INFO BlockManager: Initialized BlockManager: BlockManagerId(driver, 2a892a3469ec, 38023, None)
[2024-12-08T14:39:15.246+0000] {spark_submit.py:634} INFO - 24/12/08 14:39:15 INFO StandaloneSchedulerBackend: SchedulerBackend is ready for scheduling beginning after reached minRegisteredResourcesRatio: 0.0
[2024-12-08T14:39:15.391+0000] {spark_submit.py:634} INFO - Traceback (most recent call last):
[2024-12-08T14:39:15.393+0000] {spark_submit.py:634} INFO - File "/usr/local/lib/python3.12/urllib/parse.py", line 988, in urlencode
[2024-12-08T14:39:15.393+0000] {spark_submit.py:634} INFO - raise TypeError
[2024-12-08T14:39:15.394+0000] {spark_submit.py:634} INFO - TypeError
[2024-12-08T14:39:15.399+0000] {spark_submit.py:634} INFO - 
[2024-12-08T14:39:15.399+0000] {spark_submit.py:634} INFO - The above exception was the direct cause of the following exception:
[2024-12-08T14:39:15.402+0000] {spark_submit.py:634} INFO - 
[2024-12-08T14:39:15.403+0000] {spark_submit.py:634} INFO - Traceback (most recent call last):
[2024-12-08T14:39:15.403+0000] {spark_submit.py:634} INFO - File "/opt/***/jobs/python/wordcountjob.py", line 71, in <module>
[2024-12-08T14:39:15.404+0000] {spark_submit.py:634} INFO - file_path_clean = urllib.parse.urlencode(ftp_path, safe='|')
[2024-12-08T14:39:15.405+0000] {spark_submit.py:634} INFO - ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2024-12-08T14:39:15.406+0000] {spark_submit.py:634} INFO - File "/usr/local/lib/python3.12/urllib/parse.py", line 994, in urlencode
[2024-12-08T14:39:15.406+0000] {spark_submit.py:634} INFO - raise TypeError("not a valid non-string sequence "
[2024-12-08T14:39:15.407+0000] {spark_submit.py:634} INFO - TypeError: not a valid non-string sequence or mapping object
[2024-12-08T14:39:15.524+0000] {spark_submit.py:634} INFO - 24/12/08 14:39:15 INFO SparkContext: Invoking stop() from shutdown hook
[2024-12-08T14:39:15.525+0000] {spark_submit.py:634} INFO - 24/12/08 14:39:15 INFO SparkContext: SparkContext is stopping with exitCode 0.
[2024-12-08T14:39:15.538+0000] {spark_submit.py:634} INFO - 24/12/08 14:39:15 INFO SparkUI: Stopped Spark web UI at http://2a892a3469ec:4042
[2024-12-08T14:39:15.543+0000] {spark_submit.py:634} INFO - 24/12/08 14:39:15 INFO StandaloneSchedulerBackend: Shutting down all executors
[2024-12-08T14:39:15.551+0000] {spark_submit.py:634} INFO - 24/12/08 14:39:15 INFO StandaloneSchedulerBackend$StandaloneDriverEndpoint: Asking each executor to shut down
[2024-12-08T14:39:15.585+0000] {spark_submit.py:634} INFO - 24/12/08 14:39:15 INFO MapOutputTrackerMasterEndpoint: MapOutputTrackerMasterEndpoint stopped!
[2024-12-08T14:39:15.655+0000] {spark_submit.py:634} INFO - 24/12/08 14:39:15 INFO MemoryStore: MemoryStore cleared
[2024-12-08T14:39:15.656+0000] {spark_submit.py:634} INFO - 24/12/08 14:39:15 INFO BlockManager: BlockManager stopped
[2024-12-08T14:39:15.666+0000] {spark_submit.py:634} INFO - 24/12/08 14:39:15 INFO BlockManagerMaster: BlockManagerMaster stopped
[2024-12-08T14:39:15.670+0000] {spark_submit.py:634} INFO - 24/12/08 14:39:15 INFO OutputCommitCoordinator$OutputCommitCoordinatorEndpoint: OutputCommitCoordinator stopped!
[2024-12-08T14:39:15.689+0000] {spark_submit.py:634} INFO - 24/12/08 14:39:15 INFO SparkContext: Successfully stopped SparkContext
[2024-12-08T14:39:15.691+0000] {spark_submit.py:634} INFO - 24/12/08 14:39:15 INFO ShutdownHookManager: Shutdown hook called
[2024-12-08T14:39:15.692+0000] {spark_submit.py:634} INFO - 24/12/08 14:39:15 INFO ShutdownHookManager: Deleting directory /tmp/spark-58ee7e63-0f03-4206-a845-100a72dc2845
[2024-12-08T14:39:15.699+0000] {spark_submit.py:634} INFO - 24/12/08 14:39:15 INFO ShutdownHookManager: Deleting directory /tmp/spark-58ee7e63-0f03-4206-a845-100a72dc2845/pyspark-e0830d1a-a9ea-4a5c-ab9e-eac35b37ac42
[2024-12-08T14:39:15.707+0000] {spark_submit.py:634} INFO - 24/12/08 14:39:15 INFO ShutdownHookManager: Deleting directory /tmp/spark-e161cfd5-588e-4739-bd85-fbead56dcd34
[2024-12-08T14:39:15.832+0000] {taskinstance.py:3311} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 417, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 176, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 560, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://sparkingflow-spark-master-1:7077 --name arrow-spark --deploy-mode client jobs/python/wordcountjob.py. Error code is: 1.
[2024-12-08T14:39:15.844+0000] {taskinstance.py:1225} INFO - Marking task as FAILED. dag_id=sparking_flow, task_id=python_job, run_id=manual__2024-12-08T14:39:04.163267+00:00, execution_date=20241208T143904, start_date=20241208T143905, end_date=20241208T143915
[2024-12-08T14:39:15.865+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2024-12-08T14:39:15.866+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 166 for task python_job (Cannot execute: spark-submit --master spark://sparkingflow-spark-master-1:7077 --name arrow-spark --deploy-mode client jobs/python/wordcountjob.py. Error code is: 1.; 192)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3005, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 273, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3159, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3183, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 417, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 176, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 560, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://sparkingflow-spark-master-1:7077 --name arrow-spark --deploy-mode client jobs/python/wordcountjob.py. Error code is: 1.
[2024-12-08T14:39:15.892+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2024-12-08T14:39:15.895+0000] {standard_task_runner.py:217} INFO - Process not found (most likely exited), stop collecting metrics
[2024-12-08T14:39:15.940+0000] {taskinstance.py:3895} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-12-08T14:39:15.946+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
